components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -1087,6 +1089,28 @@ public static String resolveOriginUrlFromPlaceholders(String absoluteUrl) throws
         return StringUtils.replace(absoluteUrl, BASE_URL_PLACEHOLDER, basePath);
     }
 
+    /**
+     * This method use to replace placeholders with the hostname and port of URLs for the portal apps.
+     *
+     * @param absoluteUrl     The URL which need to resolve from placeholders.
+     * @return The resolved URL from placeholders.
+     * @throws URLBuilderException If any error occurs when building absolute public url without path.
+     */
+    public static String resolveOriginUrlFromPlaceholdersPortalApps(String absoluteUrl, String appName)
```

Comment:
Let's make this a generic method, by overloading resolveOriginUrlFromPlaceholders